mustache.js

```diff
@@ -45,6 +45,19 @@
     return obj != null && typeof obj === 'object' && (propName in obj);
   }
 
+  /**
+   * Safe way of detecting whether or not the given thing is a primitive and
+   * whether it has the given property
+   */
+  function primitiveHasOwnProperty (primitive, propName) {  
+    return (
+      typeof primitive !== 'object' 
+      && primitive !== undefined
```

Comment:
Additionally, for keeping "symmetry" with the `hasProperty` function, it'd be cool to also do the `!= null` check before the `typeof` one.